### PR TITLE
Fix missing data and convert to UTC to PST AB#11985

### PR DESCRIPTION
**Pull request by pagustinqt20**
# Fix missing data [AB#11985](https://dev.azure.com/qslvic/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/11985)

## Description

Fix the is missing the CollectionDateTime and ResultDateTime. This PR also include the fix for showing the datetime in UI as PST format.

If you are reviewing this PR, please adhere to the guidelines as [documented](https://github.com/bcgov/healthgateway/wiki/prguidance).

## Testing

-   [ ] Unit Tests Updated
-   [ ] Functional Tests Updated
-   [x] Not Required

### UI Changes

**NO** 

### Browsers Tested

-   [x] Chrome
-   [ ] Safari
-   [ ] Edge
-   [ ] Firefox

Provide an explanation for any test(s) not completed.

## Notes/Additional Comments

Any additional notes or comments that may be relevant. Include any specialized deployment steps here.

## Items to Review:

-   [General PR Guidelines](https://github.com/bcgov/healthgateway/wiki/PRguidance)

-   Fulfills Work Item Requirements

    -   The changes meet/implement story's acceptance criteria. Fixes identified problems if bug.
    -   Changes not directly related to the task requirements need to be documented on the PR.

-   Compilation / Tests

    -   The changes do not break compilation and/or unit or functional tests; unless other item addresses them specifically.

-   Logic Problems / Functional Behaviour

    -   The changes work as intended and do not introduce problems.

-   Performance

    -   The changes do not introduce obvious performance issues.

-   Documented Standards

    -   The changes adhere to the team's documented [Coding Standards](https://github.com/bcgov/healthgateway/wiki/standards).

-   Readability / Maintainability
    -   The changes can be easily understood/read and allow for future enhancements without major refactoring. Readability is preferred over "clever code".
    -   Reasoning for changes needs to be clearly articulated. Disagreements should be arbitrated by a third developer.

### CODE REVIEW

File: Apps/Laboratory/src/Models/PublicCovidTestRecord.cs [hunk 1]
```diff
@@ -116,6 +116,12 @@ public PublicCovidTestRecord()
         [JsonPropertyName("resultLink")]
         public string ResultLink { get; set; } = string.Empty;
 
+        /// <summary>
+        /// Gets or sets the state of the response.
+        /// </summary>
+        [JsonPropertyName("statusIndicator")]
+        public string StatusIndicator { get; set; } = string.Empty;
+
```
Comment (MikeLyttle) [inline Apps/Laboratory/src/Models/PublicCovidTestRecord.cs:124]: We don't need to add the StatusIndicator property.

**Review by MikeLyttle — CHANGES_REQUESTED** [review]
I'm not sure I understand the time bug.  I thought the issue was that PHSA was sending us times in PST but we were assuming they were in UTC.

File: Apps/Laboratory/src/Models/PublicCovidTestRecord.cs (hunk 1 above)
Comment (pagustinqt20) [reply Apps/Laboratory/src/Models/PublicCovidTestRecord.cs:124]: @MikeLyttle  The bug fix is also for missing data. we need to add CollectionDate, ResultDatetime and Statusindicator

File: Apps/WebClient/src/ClientApp/src/views/publicCovidTest.vue [hunk 2]
```diff
@@ -138,7 +138,7 @@ export default class PublicCovidTestView extends Vue {
 
     private formatDate(date: StringISODateTime): string {
         if (date) {
-            const dateWrapper = new DateWrapper(date);
+            const dateWrapper = new DateWrapper(date, { isUtc: true });
```
Comment (sslaws) [inline Apps/WebClient/src/ClientApp/src/views/publicCovidTest.vue:141]: The dates being returned from the service are in pacific are they not?

**Review by sslaws — CHANGES_REQUESTED** [review]

File: Apps/WebClient/src/ClientApp/src/views/publicCovidTest.vue (hunk 2 above)
Comment (pagustinqt20) [reply Apps/WebClient/src/ClientApp/src/views/publicCovidTest.vue:141]: @sslaws the kind is unspecified?
image

**Review by MikeLyttle — APPROVED** [review]

**Review by sslaws — APPROVED** [review]

**Pull request merged**